modules/develop/pages/data-transforms/run-transforms.adoc

```diff
@@ -0,0 +1,268 @@
+	A string `json:"a"`
+	B int    `json:"b"`
+}
+
+func csvToJsonTransform(e redpanda.WriteEvent) ([]redpanda.Record, error) {
+	// The input data is a CSV (without a header row) that is the structure of:
+	// key, a, b
+	reader := csv.NewReader(bytes.NewReader(e.Record().Value))
+	// Improve performance by reusing the result slice.
+	reader.ReuseRecord = true
+	output := []redpanda.Record{}
+	for {
+		row, err := reader.Read()
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			return nil, err
+		}
+		if len(row) != 3 {
+			return nil, errors.New("unexpected number of rows")
+		}
+		// Convert the last column into an int
+		b, err := strconv.Atoi(row[2])
+		if err != nil {
+			return nil, err
+		}
+		// Marshal our JSON value
+		f := Foo{
+			A: row[1],
+			B: b,
+		}
+		v, err := json.Marshal(&f)
+		if err != nil {
+			return nil, err
+		}
+		// Add our output record using the first column as the key.
+		output = append(output, redpanda.Record{
+			Key:   []byte(row[0]),
+			Value: v,
+		})
+
+	}
+	return output, nil
+}
+```
+
+--
+Validation filter transform::
++
+--
+```go
+import (
+	"encoding/json"
+
+	"github.com/redpanda-data/redpanda/src/go/transform-sdk"
+)
+
+// This example shows a filter that outputs only valid JSON into the
+// output topic.
+func main() {
+	redpanda.OnRecordWritten(filterValidJson)
+}
+
+func filterValidJson(e redpanda.WriteEvent) ([]redpanda.Record, error) {
+	v := []redpanda.Record{}
+	if json.Valid(e.Record().Value) {
+		v = append(v, e.Record())
+	}
+	return v, nil
+}
+```
+
+--
+====
+
+== Build and deploy the transform
+
+. Build the transform into a Wasm module with metadata: 
++
+```bash
+rpk transform build
+```
+
+. Deploy the Wasm module to your cluster. For example, with identity transform:
++
+```bash
+rpk transform deploy --input-topic=demo-1 --output-topic=demo-2
+```
+
+. Validate that your transform is running. For example:
+.. Produce a few records to the `demo-1` topic.
++
+```bash
+echo "foo\nbar" | rpk topic produce demo-1
+```
+.. Consume from the `demo-2` topic.
++
+```bash
+rpk topic consume demo-2
+```
++
+[,json,role="no-copy"]
+----
+{
+  "topic": "demo-2",
+  "value": "foo",
+  "timestamp": 1687545891433,
+  "partition": 0,
+  "offset": 0
+}
+
+  "topic": "demo-2",
+  "value": "bar",
+  "timestamp": 1687545892434,
+  "partition": 0,
+  "offset": 1
+}
+----
+
+NOTE: You can see `stdout` and `stderr` for your function in the broker's logs. 
+
+== Monitor data transforms
+
+You can monitor your transforms with the following metrics:
+
+* `transform_execution_latency_sec`
+* `transform_execution_errors`
+* `wasm_engine_cpu_seconds_total`
+* `wasm_engine_memory_usage`
+* `wasm_engine_max_memory`
+* `wasm_binary_executable_memory_usage`
+* `transform_processor_read_bytes`
+* `transform_processor_write_bytes`
+* `transform_processor_lag`
+* `transform_processor_failures`
+* `transform_processor_state`
+
+See xref:reference:public-metrics-reference.adoc[]
+
+== Limitations
```

Comment:
We usually put limitations at the top as the first second-level heading so users are aware before they start.